### DISCUSSION TITLE
怎么覆盖IDE一些组件的默认样式？

### DISCUSSION THREAD

**Discussion by terrihonkai** [opensumi/core#2674] [General]
我用的是 `ide-electron` 项目，想膝盖右侧bar的class `right-tab-bar`的几个属性，有哪些方式呢？

**Comment by erha19** [comment]
@terrihonkai 由于 OpenSumi 框架内采用的局部样式，你目前可以通过 CSS 选择器进行覆盖，如：

样式名为 `.sync_btn_abcde`, 你可以用 `[class^="sync_btn_"]` 选择器进行样式修改，但修改时务必适配好亮色与暗色环境颜色，合理使用框架提供的默认 Token 值，可以看一下这个文档 

https://opensumi.com/zh/docs/extension/develop/built-in-component#%E9%80%9A%E8%BF%87-classname-%E8%BF%9B%E8%A1%8C%E6%A0%B7%E5%BC%8F%E8%87%AA%E5%AE%9A%E4%B9%89

**Reply by terrihonkai** [reply]
json里面可以用选择器吗？还是说我得在哪个地方进行修改？

**Reply by erha19** [reply]
没太懂你的意思，结合点图说明一下你要修改的区域吧，一般就放在你自己的 CSS 内修改就行了

**Reply by terrihonkai** [reply]
![image](https://user-images.githubusercontent.com/132564832/236413278-e79c85b0-c6ab-4407-9dff-84d4fc663f9d.png)

例如我是想隐藏这块，将display改成none

目前在module的自定义模块的less下通过选择器覆盖是不行的

**Reply by erha19** [reply]
@terrihonkai 你这里涉及到修改页面默认布局，可以参考一下这个项目提供的实例 https://github.com/opensumi/app-desktop

**Reply by terrihonkai** [reply]
关键代码是这里对吗 https://github.com/opensumi/app-desktop/blob/main/src/editor/modules/top-tab/renderer.tsx

**Reply by terrihonkai** [reply]
@erha19 还有个问题，按照这里这样覆盖之后，原有左下角的设置icon也没了，这个该怎么找回呢
